File: Intervals/LC_252.cpp
// Written by: Jian-Yuan Lin
// Date: 2022.11.30

class Solution {
public:
    bool canAttendMeetings(vector<vector<int>>& intervals) {
        if (intervals.size() <= 1){
            return true;
        }
        sort(intervals.begin(), intervals.end(), [](const auto& a, const auto& b){
            return a[0] < b[0];
        });
        vector<vector<int>> res;
        res.push_back(intervals[0]);
        for (int i = 1; i < intervals.size(); i++){
            if (res.back()[1] > intervals[i][0]){
                return false;
            }
            res.push_back(intervals[i]);
        }
        return true;
    }
};

File: Intervals/LC_252.py
# Written by: Jian-Yuan Lin
# Date: 2022.11.30

class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) -> bool:
        intervals.sort(key = lambda x:x[0])
        def is_overlap(p1, p2):
            return p1[1] > p2[0]
        
        for i in range(len(intervals) - 1):
            if is_overlap(intervals[i], intervals[i+1]):
                return False
        return True